frontend/app/src/MetricsManager.ts

```diff
@@ -113,6 +122,42 @@ export class MetricsManager {
     this.appHash = appHash
   }
 
+  // Set metrics url if sent by the host_config, otherwise fallback to retrieving the default
+  public setMetricsConfig = (metricsUrl = ""): void => {
+    // Don't send metrics if explicitly set to "off"
+    if (metricsUrl === "off") {
+      return
+    } else if (metricsUrl) {
+      this.metricsUrl = metricsUrl
+    } else {
+      this.requestDefaultMetricsConfig()
+    }
+  }
+
+  // Fallback - Checks if cached in localStorage, otherwise fetches the config from a default URL
+  private async requestDefaultMetricsConfig(): Promise<any> {
+    const isLocalStoreAvailable = localStorageAvailable()
+
+    if (isLocalStoreAvailable) {
+      const cachedConfig = localStorage.getItem("stMetricsConfig")
+      if (cachedConfig) {
+        this.metricsUrl = cachedConfig
+        return
+      }
+    }
+
+    const response = await fetch(DEFAULT_METRICS_CONFIG)
+    if (!response.ok) {
+      throw new Error(`Failed to fetch metrics config: ${response.status}`)
```

Comment:
What happens if this error is thrown? Is the app still usable? Maybe its better to just log the error and set metricsUrl to undefined to keep the app functional in this case.

Reply:
The app is still usable, error just appears in the console like so:
<img width="465" alt="Screenshot 2024-10-09 at 4 47 32 PM" src="https://github.com/user-attachments/assets/cb5fa098-3250-4b9a-bd4c-fd509f4bb2c2">

But agreed makes more sense to just explicitly log error & set to undefined 👍🏼